src/btree/bt_page.c

```diff
@@ -624,6 +625,11 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page)
                  * from the data store, when the prepared transaction gets rollback.
                  */
                 if (WT_TIME_WINDOW_HAS_STOP(&unpack.tw)) {
+                    /*
+                     * Increase the total allocation cache size before the size gets overwritten
+                     * with the tombstone allocation size.
+                     */
+                    total_size += size;
```

Comment:
what if the memory allocation fails in the next statement.

Reply:
Already allocated memory gets freed during error.

Reply:
I mean the total_size is not decremented in case the memory allocation fails, so better to increment after the allocation.

Reply:
There is no need to decrement the total_size. If the allocation fails, the calculated value until now is completely ignored and error is raised.

Reply:
Yes, I understand for consistency sake, better to move the size increment after the allocation.

Reply:
Fixed by incrementing the size next after the allocation.